libs/agno/agno/knowledge/reader/base.py

```diff
@@ -40,13 +41,45 @@ def __init__(
         self.description = description
         self.max_results = max_results
 
+    def set_chunking_strategy_from_string(self, strategy_name: str, **kwargs) -> None:
+        """Set the chunking strategy from a string name."""
+        try:
+            strategy_type = ChunkingStrategyType.from_string(strategy_name)
+            self.chunking_strategy = ChunkingStrategyFactory.create_strategy(strategy_type, **kwargs)
+        except ValueError as e:
+            raise ValueError(f"Failed to set chunking strategy: {e}")
+
     def read(self, obj: Any, name: Optional[str] = None) -> List[Document]:
         raise NotImplementedError
 
     async def async_read(self, obj: Any, name: Optional[str] = None) -> List[Document]:
         raise NotImplementedError
 
+    @classmethod
+    def get_supported_chunking_strategies(cls) -> List[ChunkingStrategyType]:
+        """Get the list of supported chunking strategies for this reader.
+
+        Returns:
+            List of ChunkingStrategyType enums that this reader supports.
+        """
+        # Default implementation returns common strategies
+        return [
```

Comment:
Shouldn't have a default. By default a reader shouldn't support everything. This should be not-implemented